agent_code/auto_bomber/model.py

```diff
@@ -32,19 +34,27 @@ def select_best_action(self, game_state: dict, agent_self):
         choice = np.random.choice(top_3_actions, p=[0.9, 0.05, 0.05])
         return config.ACTIONS[choice]
 
-    def fit_model_with_transition_batch(self, transitions: Transitions):
+    def fit_model_with_transition_batch(self, transitions: Transitions, round: int):
+        loss = []
+        numpy_transitions = transitions.to_numpy_transitions()
         for action_id, action in enumerate(config.ACTIONS):
-            numpy_transitions = transitions.to_numpy_transitions()
             x_all_t, y_all_t = numpy_transitions.get_features_and_value_estimates(action)
 
             if x_all_t.size != 0:
                 q_estimations = np.sum(x_all_t * self.weights[action_id], axis=0)
                 residuals = (y_all_t - q_estimations[:, np.newaxis])
+                mean_residuals = np.mean(residuals)
+                loss.append(mean_residuals)
                 q_grad = np.sum(x_all_t.transpose() * residuals, axis=1)
 
                 weight_updates = config.LEARNING_RATE / y_all_t.shape[0] * q_grad
                 self.weights[action_id] += weight_updates
 
+        loss = np.mean(loss)
```

Comment:
This redefines/reassigns the variable loss with a other shape/type or? (Before it was a list now a scalar or?) Thus I would recommend to use another var name